File: WebVersion/Code/ThreadAction.ts
class ThreadAction {
    public threadID: number;
    public sourceIndex: number;
    public destIndex: number;
    public count: number;

    constructor(threadID: number, sourceIndex: number, destIndex: number, count: number) {
        this.threadID = threadID;
        this.sourceIndex = sourceIndex;
        this.destIndex = destIndex;
        this.count = count;
    }


}

File: SharedMemLib/ThreadAction.cs
namespace SharedMemLib
{
    public class ThreadAction
    {
        public int ThreadId { get; }
        public int SourceIndex { get; }
        public int DestIndex { get; }
        public int Count { get; }

        public ThreadAction(int threadId, int sourceIndex,int destIndex, int count)
        {
            ThreadId = threadId;
            SourceIndex = sourceIndex;
            DestIndex = destIndex;

            Count = count;
        }

        public override string ToString()
        {
            return $"[{ThreadId}] : {SourceIndex}->{DestIndex} ({Count})";
        }
    }
}
